agent/src/main/java/com/cloud/agent/AgentShell.java

```diff
@@ -399,12 +401,17 @@ public void init(String[] args) throws ConfigurationException {
         _backoff.configure("ConstantTimeBackoff", new HashMap<String, Object>());
     }
 
-    private void enableSSL() {
+    private void enableSSLForKvmAgent() {
         final File agentFile = PropertiesUtil.findConfigFile("agent.properties");
         if (agentFile == null) {
             s_logger.info("Failed to find agent.properties file");
             return;
         }
+        final String resource = getProperty(null, "resource");
+        if (!LIBVIRT_COMPUTING_RESOURCE.equalsIgnoreCase(resource)) {
```

Comment:
The use of `LIBVIRT_COMPUTING_RESOURCE` or `com.cloud.hypervisor.kvm.resource.LibvirtComputingResource` may be misleading (for ex. what if this isn't the resource name? for example, we do support lxc etc?).

Can you simply add a check if the agent is either is in systemvm (for ex. /etc/cloudstack-release would exist) or if we're on a KVM host (check for /dev/kvm exists or libvirt running)? (Depending on what we're trying to achieve here). The other option usually is to check the resource class via some method.

What do you think @weizhouapache @nvazquez ?

Reply:
@rohityadavcloud 
I will move all these codes to `LibvirtComputingResource` which makes more sense.
I will verify the issue fixed by #6200 as well.

Reply:
+1